compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -954,6 +954,29 @@ pub(crate) unsafe fn codegen(
     ))
 }
 
+fn create_section_with_flags_asm(section_name: &str, section_flags: &str, data: &[u8]) -> Vec<u8> {
+    let mut asm = format!(".section {},\"{}\"\n", section_name, section_flags).into_bytes();
+    asm.extend_from_slice(b".ascii \"");
+    asm.reserve(data.len());
+    for &byte in data {
+        if byte == b'\\' || byte == b'"' {
+            asm.push(b'\\');
+            asm.push(byte);
+        } else if byte < 0x20 || byte >= 0x80 {
```

Comment:
To be honest, I'm somewhat more comfortable with a `.byte` directive over `.ascii` with escapes here. Is there any particular reason why we'd be stuck with `.ascii`?

Reply:
`.byte` should work as well. I picked `.ascii` here for more compact encoding. Though I guess `.byte` isn't so bad when using decimal rather than hex and saving on the `0x` prefix for each byte.